File: server/src/main/java/tn/JobPortal/RestEasy/Model/Offer.java
package tn.JobPortal.RestEasy.Model;

import jakarta.nosql.mapping.Column;
import jakarta.nosql.mapping.Entity;
import jakarta.nosql.mapping.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity("Offer")
public class Offer {
    @Column
    private String title;
    @Column
    private String dateOfPub;
    @Id
    @Column
    private String originalLink;
    @Column
    private String company;
    @Column
    private String location;
}

File: scripts/Offer.ts
export class Offer {
    title: string;
    dateOfPub: string;
    company: string;
    location: string;
    originalLink: string;
    constructor(
        job: string,
        pub: string,
        comp: string,
        loc: string,
        infos: string
    ) {
        this.title = job;
        this.dateOfPub = pub;
        this.company = comp;
        this.location = loc;
        this.originalLink = infos;
    }
}